pkg/build/builder/docker.go

```diff
@@ -487,3 +501,50 @@ func getDockerfileFrom(dockerfilePath string) []string {
 	}
 	return froms
 }
+
+// rewriteDockerfileForImages returns all qualified images referenced by the Dockerfile, whether the
+// build is a multi-stage build, or returns an error.
+// TODO: implement image rewriting (so that multiple replacements can be made in multi-stage image).
+func rewriteDockerfileForImages(dockerfilePath string, build *buildapiv1.Build) ([]string, bool, error) {
```

Comment:
why is this called "rewriteDockerfile"?  it doesn't appear to be rewriting anything, just returning the referenced images.

I see you have a TODO to handle rewriting but i don't think this is where we'd want to implement it.  We handle dockerfile rewriting (including updating the FROM) here: 
https://github.com/openshift/origin/blob/master/pkg/build/builder/source.go#L94-L125